Content.Server/DeviceLinking/Systems/SignallerSystem.cs

```diff
@@ -48,4 +55,10 @@ private void OnTrigger(EntityUid uid, SignallerComponent component, TriggerEvent
         _link.InvokePort(uid, component.Port);
         args.Handled = true;
     }
+
+    private void OnSignalFailed(EntityUid uid, SignallerComponent component, ref SignalFailedEvent args) // called by WirelessNetworkSystem
+    {
+        // makes a popup if the signaller fails to trigger
+        _popup.PopupEntity(Loc.GetString("signaller-interact-failed-to-activate-text"), uid, component.LastUser, PopupType.SmallCaution);
+    }
```

Comment:
instead of storing the last person who used it, just get the person holding the item through ContainerSystem